tensorboard/components/tensor_widget/tensor-widget-impl.ts

```diff
@@ -146,12 +178,388 @@ export class TensorWidgetImpl implements TensorWidget {
     this.infoSubsection.appendChild(shapeTagDiv);
   }
 
+  /**
+   * Fill in the content of the value divs given the current slicing spec.
+   */
+  private async renderValues() {
+    if (this.valueSection == null) {
+      this.valueSection = document.createElement('div');
+      this.rootElement.appendChild(this.valueSection);
+
+      this.valueSection.addEventListener('wheel', async (event) => {
+        event.stopPropagation();
+        event.preventDefault();
+        await this.scrollUpOrDown(event.deltaY > 0 ? 'down' : 'up');
+      });
+    }
+    // TOOD(cais): Remove this check once 2D+ tensors are supported.
+    if (this.rank <= 2) {
+      this.createTopRuler();
+      this.createLeftRuler();
+      this.createValueDivs();
+      await this.renderRulersAndValueDivs();
+    }
+  }
+
+  /**
+   * Creates the top ruler.
+   *
+   * The top ruler includes the topleft-most ruler block, in addition to the
+   * column-wise ruler blocks.
+   */
+  private createTopRuler() {
+    if (this.topRuler == null) {
+      this.topRuler = document.createElement('div');
+      this.topRuler.classList.add('tenesor-widget-top-ruler');
+      this.valueSection.appendChild(this.topRuler);
+      this.topRulerTicks = [];
+
+      this.topRuler.addEventListener('wheel', async (event) => {
```

Comment:
If I didn't see this code I wouldn't have known you can scroll left right using the wheel here.  Is there some sort of visual affordance you can offer?  Possibly change the cursor to <-> on hover?